incubator/solr/templates/statefulset.yaml

```diff
@@ -63,13 +63,20 @@ spec:
             - |
               COUNTER=0;
               while [  $COUNTER -lt 120 ]; do
-                for i in {{ $vals := . -}}{{ range $i, $e := until ( int .Values.zookeeper.replicaCount ) -}}
-                  "{{- include "solr.zookeeper-name" $vals }}-{{ $i }}.{{ include "solr.zookeeper-service-name" $vals }}" {{ end -}};
-                  do mode=$(echo srvr | nc $i 2181 | grep "Mode");
-                    if [ "$mode" == "Mode: leader" ] || [ "$mode" == "Mode: standalone" ]; then
+                addr=$(nslookup -type=a {{ include "solr.zookeeper-service-name" . }} | grep "Address:" | awk 'NR>1 {print $2}')
```

Comment:
I'm concerned the use of `nslookup` might break people that are using custom docker images without nslookup in them.

Reply:
I understand your concern, but this affects only the init container which is using busybox:latest and not the main Solr docker image. I hope this is what you meant.

Besides, there's currently no field to change the init container image in values.yaml and I think people should be "forced" to use an init container with nslookup or remove the init container completely.

Reply:
I understand your concern, but this only affects the init container which uses busybox:latest and not the main solr container which I presume that some people would swap for a custom one. I hope this is what you were thinking.

I have tried getent but busybox it does not support it. Besides, there is no field for a user to change the init container image from values.yaml and I think it should be "forced" to use an init container with nslookup or go without an init container at all.